worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/FileWriter.java

```diff
@@ -159,14 +162,24 @@ protected void flush(boolean finalFlush) throws IOException {
         int numBytes = flushBuffer.readableBytes();
         if (numBytes != 0) {
           notifier.checkException();
-          notifier.numPendingFlushes.incrementAndGet();
-          FlushTask task = null;
-          if (channel != null) {
-            task = new LocalFlushTask(flushBuffer, channel, notifier);
-          } else if (fileInfo.isHdfs()) {
-            task = new HdfsFlushTask(flushBuffer, fileInfo.getHdfsPath(), notifier);
+          if (finalFlush
+              && channel != null
+              && fileInfo.getFileLength() == 0
+              && cacheManager.canCache(numBytes)) {
+            cacheManager.putCache(fileInfo.getFilePath(), flushBuffer.copy());
```

Comment:
Actually, this copy is unnecessary. You can just use this buffer and return it when cache is released.

Reply:
copy() here is to simplify release logic